src/app/courses-page/services/courses.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+
+import { Course } from '@app/models';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CoursesService {
+  courses: Course[] = [
+    {
+      id: '1',
+      title: 'Video Course 1. Name tag',
+      creationDate: new Date('2020, 11, 16'),
+      duration: 40,
+      description:
+        'Learn about where you can find course descriptions, what information they include, how they work, and details about various components of a course description. Course descriptions report information about a university or college\'s classes. They\'re published both in course catalogs that outline degree requirements and in course schedules that contain descriptions for all courses offered during a particular semester.',
+      topRated: true
+    },
+    {
+      id: '2',
+      title: 'Video Course 2. Name tag',
+      creationDate: new Date('2020, 08, 28'),
+      duration: 120,
+      description:
+        'Learn about where you can find course descriptions, what information they include, how they work, and details about various components of a course description. Course descriptions report information about a university or college\'s classes. They\'re published both in course catalogs that outline degree requirements and in course schedules that contain descriptions for all courses offered during a particular semester.',
+      topRated: true
+    },
+    {
+      id: '3',
+      title: 'Video Course 3. Name tag',
+      creationDate: new Date('2020, 11, 20'),
+      duration: 98,
+      description:
+        'Learn about where you can find course descriptions, what information they include, how they work, and details about various components of a course description. Course descriptions report information about a university or college\'s classes. They\'re published both in course catalogs that outline degree requirements and in course schedules that contain descriptions for all courses offered during a particular semester.'
+    }
+  ];
+
+  constructor() { }
+
+  getCourses(): Course[] {
+    return this.courses;
+  }
+
+  createCourse(course: Course): void {
+    this.courses.push(course);
+  }
+
+  getCourse(id: string): Course {
+    return this.courses.find(course => course.id === id);
+  }
+
+  updateCourse(course: Course): void {
+    const updateIndex = this.courses.findIndex(item => item.id === course.id);
+    this.courses[updateIndex] = course; 
```

Comment:
better to return new array of courses. In that case angular will update bindings if you have `onPush` strategy. You can use it in the combination with [`trackBy`](https://netbasal.com/angular-2-improve-performance-with-trackby-cc147b5104e5) function to optimize performance

Reply:
done